script/CheckSecurityConfigs.s.sol

```diff
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { console2 } from "forge-std/console2.sol";
+import { Script } from "forge-std/Script.sol";
+import { stdStorage } from "forge-std/StdStorage.sol";
+
+/**
+ * @title CheckSecuityConfigs
+ * @notice A script to check security configurations of an OP Chain,
+           such as upgrade key holder, challenger and guadian designations.
+ *         The usage is as follows:
+ *         $ forge script CheckSecuityConfigs \
+ *             --rpc-url $ETH_RPC_URL
+ */
+
+contract CheckSecuityConfigs is Script {
+    struct ProtocolAddresses {
+        // Protocol contracts
+        address AddressManager;
+        address L1CrossDomainMessengerProxy;
+        address L1ERC721BridgeProxy;
+        address L1StandardBridgeProxy;
+        address L2OutputOracleProxy;
+        address OptimismMintableERC20FactoryProxy;
+        address OptimismPortalProxy;
+        address ProxyAdmin;
+        address SystemConfigProxy;
+
+        // Privileged roles
+        address Challenger;
+        address Guardian;
+        address ProxyAdminOwner;
+        address SystemConfigOwner;
+    }
+
+    bool internal hasErrors;
+
+    /**
+     * @notice The entrypoint function.
+     */
+    function run() external {
+        string[4] memory addressesJsonFiles = [
+            "superchain/extra/addresses/mainnet/base.json",
+            "superchain/extra/addresses/mainnet/op.json",
+            "superchain/extra/addresses/mainnet/pgn.json",
+            "superchain/extra/addresses/mainnet/zora.json"
+        ];
```

Comment:
Is there no way to iterate all files in the directory, instead of hardcoding them? Maybe retrieve a dir listing with FFI?